lib/reports/subscriber_lists_report.rb

```diff
@@ -0,0 +1,69 @@
+class Reports::SubscriberListsReport
+  attr_reader :date
+
+  CSV_HEADERS = %i[title
+                   slug
+                   matching_criteria
+                   created_at
+                   individual_subscribers
+                   daily_subscribers
+                   weekly_subscribers
+                   matched_content_changes_for_date
+                   matched_messages_for_date
+                   unsubscribed].freeze
+
+  def initialize(date)
+    @date = Time.zone.parse(date)
+  end
+
+  def call
+    CSV do |csv|
+      csv << CSV_HEADERS
+
+      SubscriberList.where("created_at < ?", date.end_of_day).find_each do |list|
+        csv << export_list_row(list)
+      end
+    end
+  end
+
+private
+
+  def export_list_row(list)
+    unsubscribed_count = list.subscriptions.where("ended_at <= ?", date.end_of_day).count
```

Comment:
If we're keeping this, we should clarify that it refers to "unsubscriptions" (over all time). Based on [our discussion](https://github.com/alphagov/email-alert-api/pull/1456#discussion_r513576736), I actually think it would be more useful to say "unsubscribes_for_date" or similar.

Reply:
I think it would be worse to add 'unsubscribes_for_date' as the "for_date" suffix is used for matched content changes / messages to mean **only** on a specific date.  

You could do something like "for_all_time" but this isn't strictly true either due to our data retention policy.

I think just 'unsubscribed' is okay as it follows the same pattern as individual/daily/weekly subscribers by not having any time related suffix. I could update them all so we are consistent if you can think of a viable option?

Reply:
> is used for matched content changes / messages to mean only on a specific date.

Indeed. I was being a bit implicit there: my suggestion was to change the behaviour to be like that, which I imagine is more useful in practice: all time unsubscribes is a pretty useless metric by itself.

Alternatively, the wording could change to make it clearer this is a counter: using the noun ("unsubscriptions") instead of an unqualified adjective ("unsubscribed" ... when?).

Up to you - I think the first option is more useful.

Reply:
> Indeed. I was being a bit implicit there: my suggestion was to change the behaviour to be like that, which I imagine is more useful in practice: all time unsubscribes is a pretty useless metric by itself.

I think it's more useful as 'all time' as you could extract more information from it. Example being if you wanted the unsubscribers for a day then you could just run the task twice over two days and you'd see the difference. The benefit of counting 'all' unsubscriptions is you could see the amount between a larger range whereas you can't if you only have a "for_date" method unless you wanted to run it for every single day. I suppose it depends how you use the report to which is more useful.

> Alternatively, the wording could change to make it clearer this is a counter: using the noun ("unsubscriptions") instead of an unqualified adjective ("unsubscribed" ... when?).

Good point, I've updated the name to your suggestion.

> Up to you - I think the first option is more useful.

Appears we agree to disagree 😄 Doesn't happen often! If daily unsubscribes turns out to be wanted then I have no qualms updating the code.

Reply:
> Example being if you wanted the unsubscribers for a day then you could just run the task twice over two days and you'd see the difference

This is amusing: option 1 does exactly that, but with one step 😆.

> I suppose it depends how you use the report to which is more useful.

Yes, that's true. I'm struggling to come to terms with "total number of unsubscriptions over all time", just because it's not a quantity we've ever referred to before. However, we can also easily change it at the moment.